packages/sanity/src/_internal/cli/commands/backup/downloadBackupCommand.ts

```diff
@@ -0,0 +1,288 @@
+import {defaultApiVersion} from './backupGroup'
+
+const debug = require('debug')('sanity:backup')
+
+const DEFAULT_DOWNLOAD_CONCURRENCY = 10
+const MAX_DOWNLOAD_CONCURRENCY = 24
+
+type DownloadBackupOptions = {
+  projectId: string
+  datasetName: string
+  token: string
+  backupId: string
+  outDir: string
+  outFileName: string
+  overwrite: boolean
+  concurrency: number
+}
+
+const helpText = `
+Options
+  --backup-id <string> The backup ID to download. (required)
+  --out <string>       The file or directory path the backup should download to.
+  --overwrite          Allows overwriting of existing backup file.
+  --concurrency <num>  Concurrent number of backup item downloads. (max: 24)
+
+Examples
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-1
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-2 --out /path/to/file
+  sanity backup download DATASET_NAME --backup-id 2024-01-01-backup-3 --out /path/to/file --overwrite
+`
+
+const downloadBackupCommand: CliCommandDefinition = {
+  name: 'download',
+  group: 'backup',
+  signature: '[DATASET_NAME]',
+  description: 'Download a dataset backup to a local file.',
+  helpText,
+  // eslint-disable-next-line max-statements
+  action: async (args, context) => {
+    const {output, chalk} = context
+    const [client, opts] = await prepareBackupOptions(context, args)
+    const {projectId, datasetName, backupId, outDir, outFileName} = opts
+
+    // If any of the output path or file name is empty, cancel the operation.
+    if (outDir === '' || outFileName === '') {
+      output.print('Operation cancelled.')
+      return
+    }
+    const outFilePath = path.join(outDir, outFileName)
+
+    output.print('╭───────────────────────────────────────────────────────────╮')
+    output.print('│                                                           │')
+    output.print('│ Downloading backup for:                                   │')
+    output.print(`│ ${chalk.bold('projectId')}: ${chalk.cyan(projectId).padEnd(56)} │`)
+    output.print(`│ ${chalk.bold('dataset')}: ${chalk.cyan(datasetName).padEnd(58)} │`)
+    output.print(`│ ${chalk.bold('backupId')}: ${chalk.cyan(backupId).padEnd(56)} │`)
+    output.print('│                                                           │')
+    output.print('╰───────────────────────────────────────────────────────────╯')
+    output.print('')
+    output.print(`Downloading backup to "${chalk.cyan(outFilePath)}"`)
+
+    const start = Date.now()
+    const progressSpinner = newProgress(output, 'Setting up backup environment...')
+
+    // Create a unique temporary directory to store files before bundling them into the archive at outputPath.
+    // Temporary directories are normally deleted at the end of backup process, any unexpected exit may leave them
+    // behind, hence it is important to create a unique directory for each attempt.
+    const tmpOutDir = await mkdtemp(path.join(tmpdir(), `backup-`))
```

Comment:
would be great if we could add `sanity-` to the tempfile here so we can leave a hint about who made it.

Reply:
Goood point. Added it.